modules/slugs/includes/qtranslate-slug-settings-options.php

```diff
@@ -47,15 +47,27 @@ function qts_options_page_fields() {
         $options[] = qts_options_page_build_slug_fields( $taxonomy, "taxonomies", "taxonomy_" );
     }
 
-    return $options;
+    return array_filter($options);
 }
 
 function qts_options_page_build_slug_fields( $object, $target_section, $id_prefix ) {
+    if ( ! is_array($object->rewrite) ) {
+        if ( $object->rewrite ){
+            $slug = $object->name;
+        }else{
+            return array();
+        }
+    } else if (array_key_exists( 'slug', $object->rewrite ) ) {
+        $slug = ltrim( $object->rewrite['slug'], "/" );
+    } else {
+        $slug = $object->name;
+    }
+        
```

Comment:
Could we have objects that don't have `rewrite`?

Reply:
yes e.g. for posts and pages `rewrite==false` meaning that permalink bases cannot be modified. `rewrite`property can be boolean or array. as long as `rewrite!=false` permalink base is ok to be customized

Reply:
last `else `is covering the case `rewrite `is an array but no `slug `key is there (still ok to be customized)